fitbenchmarking/parsing/parse.py

```diff
@@ -41,6 +42,14 @@ def parse_problem_file(prob_file):
 
     logger.info("* Testing fitting of problem {0}".format(problem.name))
 
+    if problem.sort == True:
+        index = argsort(problem.data_x)
+        problem.data_x = problem.data_x[index]
+        problem.data_y = problem.data_y[index]
+        if isinstance(problem.data_e, ndarray):
+            problem.data_e = problem.data_e[index]
+
```

Comment:
What if `sort==False`: if this for when we know the data is pre-sorted?

If this is meant for a user (or someone putting in a new model) to change, then we should mention it in the docs somewhere -- even if it just hangs there for now.  Otherwise, I think we should just sort by default, and add in a switch if and when we think it's necessary.

Reply:
Lets add the switch later when we have done the cleaning in the problem definitions